### PR TITLE
Wiki article about #include

### PR DESCRIPTION
**Pull request by nevemlaci**
Talked about this yesterday, tried my best :'D

### CODE REVIEW

File: articles/include.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<!-- alias include -->
+
+# Including headers
+## What is [`#include`](https://en.cppreference.com/w/cpp/preprocessor/include)?
```
Comment (jeremy-rifkin) [inline articles/include.md:4]: Unfortunately discord doesn't support links in embed field headers, at least not like this

File: articles/include.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+## Include paths
+`#include <...>` looks up the "include paths" that are specified by the system and compiler. You can add paths to this by:<br>
+Passing `-I "some/directory/"` on gcc/clang, `/I "some/directory/"` on MSVC.<br>
+Adding `target_include_directories(targetname PRIVATE some/directory/)` in CMake
+
+`#include "..."` looks up the directory relative to the source file it's in. If it didn't find the header file that way, it looks up the include paths as a fallback.
+
+## Misconceptions:
+Header files have the minimal information necessary for the compiler at compile time.<br>
```
Comment (jeremy-rifkin) [inline articles/include.md:15]: Maybe instead of this section it'd be better to explain why forward declarations of functions are a thing and why #include can cause linker errors? (and then the next section can merge with that)

File: articles/include.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
+<!-- alias include -->
+
+# Including headers
+## What is [`#include`](https://en.cppreference.com/w/cpp/preprocessor/include)?
+`#include` includes a header file into the current file at the line immediately after the directive. This is practically a copy paste done by the preprocessor.
```
Comment (jeremy-rifkin) [inline articles/include.md:5]: It might be good to shorten this to "`#include` is used to include code from header files. Under the hood, `#include` is essentially a copy-paste of file contents.", or similar

**Review by jeremy-rifkin — COMMENTED** [review]
Thanks so much for taking the time to put this together, I have a few quick comments/suggestions

File: articles/include.md (hunk 2 above)
Comment (nevemlaci) [reply articles/include.md:15]: What do you mean by that? I'm not sure what you mean by linker errors caused by `#include`
I can explain why circular includes don't work briefly if that is what you meant by that.

Comment (nevemlaci) [reply articles/include.md:15]: Also I don't see how I can add forward declarations in there without taking out relevant information and not making the article too long :'D

**Comment by nevemlaci** [comment]
Made some changes based on the suggestions

File: articles/include.md (hunk 2 above)
Comment (jeremy-rifkin) [reply articles/include.md]: What I was thinking is stuff like `void foo() {}` in a header causing multiple definition linker errors. I think this would be useful to highlight as it can demonstrate the reason why headers generally have only forward-declarations. In the context of headers being a copy-paste, I think this does a good job explaining why headers generally only have declarations and why you shouldn't #include source files.

Comment (nevemlaci) [reply articles/include.md]: added

**Review by jeremy-rifkin — APPROVED** [review]
Thanks again for taking the time to do this! I'm happy to merge and we can keep iterating as needed.
